src/mutations/article/editArticle.ts

```diff
@@ -219,9 +220,47 @@ const resolver: MutationToEditArticleResolver = async (
         limit: null,
       })
     ).map(({ articleId }: { articleId: string }) => articleId)
+
     const newIds = uniq(
-      collection.map((articleId) => fromGlobalId(articleId).id)
+      (
+        await Promise.all(
+          collection.map(async (articleId) => {
+            const articleDbId = fromGlobalId(articleId).id
+
+            if (!articleDbId) {
+              return
+            }
+
+            const collectedArticle = await atomService.findUnique({
+              table: 'article',
+              where: { id: articleDbId },
+            })
+
+            if (!collectedArticle) {
+              throw new ArticleNotFoundError(`Cannot find article ${articleId}`)
+            }
+
+            if (collectedArticle.state !== ARTICLE_STATE.active) {
+              throw new ForbiddenError(
+                `Article ${articleId} cannot be collected.`
+              )
+            }
+
+            const isBlocked = await userService.blocked({
+              userId: collectedArticle.authorId,
+              targetId: viewer.id,
+            })
+
+            if (isBlocked) {
+              return
+            }
+
+            return articleDbId
+          })
+        )
+      ).filter((articleId): articleId is string => !!articleId)
```

Comment:
`articleId is string` new feature on TypeScript? 👀

Reply:
It's not a new feature. The concept is similar to `Type guard`.